frontend/src/hooks/OAuth2/useFacebookOAuth2.ts

```diff
@@ -2,23 +2,22 @@ import { useHttp } from '../Http/useHttp';
 import { useCallback, useContext, useEffect, useState } from 'react';
 import { UserContext } from '../../contexts/User/UserContext';
 import { FacebookOAuth2 } from './FacebookOAuth2';
-import { facebookParametersKey } from "./FacebookParametersKey";
+import { facebookParametersKey } from './FacebookParametersKey';
+import { useLogin } from '../UserActions/Login/useLogin';
 
 export const useFacebookOAuth2 = (): FacebookOAuth2 => {
     const { setUser } = useContext(UserContext);
     const [authorizationError, setAuthorizationError] = useState<string | undefined>();
     const [authorizationFinished, setAuthorizationFinished] = useState(false);
     const { state: authorizationState, sendRequest } = useHttp<AuthorizationResponse>();
+    const login = useLogin();
 
     useEffect(() => {
         if (authorizationState.response) {
-            setUser({
-                token: authorizationState.response.token,
-                tokenExpiresIn: authorizationState.response.expiresIn
-            });
+            login.doLogin(authorizationState.response.token, authorizationState.response.expiresIn);
```

Comment:
[nit] You can extract 'authorizationState.response' in a local variable and use it in that method.